src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IRequestExecutionQueue.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface IRequestExecutionQueue<RequestContextType> where RequestContextType : struct
```

Comment:
Do we need this, now that there is an abstract base class?

Reply:
I think that it makes sense in case people look at our queuing system and say "no, none of this". On the other hand maybe this is the equivalent of the options proliferation I'm always railing against and I'm abstracting something nobody will ever use (a queue is a queue after all). Something to discuss when we have the meeting it sounds like we're gonna have about our initial surface area.